### ISSUE TITLE
MuForce hit with GoldEd/SetPointer

### ISSUE THREAD

**Issue by shred** [shred/patchwork#1]
@Hagbard-Celin reported this MuForce hit via email:

```
*MuForce Hit
LONG READ from 120F000A                        PC: 041B8794
USP : 0472E878 SR: 0018  (U0)(F)(-)  TCB: 0471EBA0
Data: 041BB7FF 000000FF 00000001 0000FFFF 00000066 00000000 0472E890 000012FF
Addr: 041BB682 120F0000 041BB6C4 047FC1C0 041BB6B2 0471637C 04000884 040022A4
Stck: 00000000 00000000 00004D04 0000001E 000002CF 00000043 00000000 041BB74D
Stck: 00000000 00000000 00000001 0000FFFF 00000066 0000000F 0000000F 0000127F
----> 041B8794 - "PatchWork"  Hunk 0000 Offset 0000078C
----> 041BB74D - "PatchWork"  Hunk 0000 Offset 00003745
041b8774 :  4241                       clr.w d1
041b8776 :  4841                       swap.l d1
041b8778 :  0800 0003                  btst #$3,d0
041b877c :  6706                       beq.s $41b8784
041b877e :  c2bc 0000 00ff             and.l #$ff,d1
041b8784 :  0800 0004                  btst #$4,d0
041b8788 :  6702                       beq.s $41b878c
041b878a :  e589                       lsl.l #$2,d1
041b878c :  2f01                       move.l d1,-(a7)
041b878e :  60b8                       bra.s $41b8748
041b8790 :  226d 0038                  movea.l $38(a5),a1
041b8794 : *2f29 000a                  move.l $a(a1),-(a7)
041b8798 :  224f                       movea.l a7,a1
041b879a :  4eb9 041b c464             jsr $41bc464
041b87a0 :  2e46                       movea.l d6,a7
041b87a2 :  2f05                       move.l d5,-(a7)
041b87a4 :  41fa 0124                  lea.l $41b88ca(pc),a0
041b87a8 :  224f                       movea.l a7,a1
041b87aa :  4eb9 041b c464             jsr $41bc464
041b87b0 :  588f                       addq.l #$4,a7
041b87b2 :  4cdf 0102                  movem.l (a7)+,d1/a0
Name: "Background CLI"  CLI: "golded:golded"

 SetPointer(0x00000000,0x00000000,19716,30,719,67)
Severity 0: hot spot is outside of the sprite
PC=651C03FF TCB=0471EBA0 ("Background CLI")
Data: 046C4D04 0066001E 033302CF 00410043 033302CF 033302CF 2820327F 00000000
Addr: 00000000 00000000 00000000 0000042F A5600491 9250040B 120F0000 00000471
Stck: 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
Stck: 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000

*MuForce Hit
BYTE READ from 2B4000B4                        PC: 041EABB4
USP : 04A27D28 SR: 0000  (U0)(F)(-)  TCB: 04A258A8
Data: 2B4000B4 0000000F 00000001 0000FFFF 000F1A89 FFFFFFFF 00000000 00000000
Addr: 041BB682 041BB682 042A3564 2B4000B5 041BB6B2 04A27E1C 041E5CC8 040022A4
Stck: 0000FFFF 000F1A89 00000000 04A27E68 0000C0FF 04A27E4C 041BB682 042A3564
Stck: 041E5CC8 00257300 541F9E21 042A3564 041BB683 041E5CC8 042A3564 041E5CC8
----> 041EABB4 - "LIBS:locale.library"  Hunk 0000 Offset 0000482C
----> 041BB682 - "PatchWork"  Hunk 0000 Offset 0000367A
----> 041BB683 - "PatchWork"  Hunk 0000 Offset 0000367B
041eab94 :  2a6f 00d8                  movea.l $d8(a7),a5
041eab98 :  601a                       bra.s $41eabb4
041eab9a :  7000                       moveq.l #$0,d0
041eab9c :  102f 0024                  move.b $24(a7),d0
041eaba0 :  2f0a                       move.l a2,-(a7)
041eaba2 :  2240                       movea.l d0,a1
041eaba4 :  204d                       movea.l a5,a0
041eaba6 :  2c6e 005c                  movea.l $5c(a6),a6
041eabaa :  4eae ff9a                  jsr -$66(a6)
041eabae :  245f                       movea.l (a7)+,a2
041eabb0 :  2c6f 0034                  movea.l $34(a7),a6
041eabb4 : *101b                       move.b (a3)+,d0
041eabb6 :  1f40 0024                  move.b d0,$24(a7)
041eabba :  66de                       bne.s $41eab9a
041eabbc :  2c6f 0034                  movea.l $34(a7),a6
041eabc0 :  6000 00fe                  bra $41eacc0
041eabc4 :  7073                       moveq.l #$73,d0
041eabc6 :  b02f 0026                  cmp.b $26(a7),d0
041eabca :  2c6f 0034                  movea.l $34(a7),a6
041eabce :  6610                       bne.s $41eabe0
041eabd0 :  200b                       move.l a3,d0
041eabd2 :  670c                       beq.s $41eabe0
```

**Comment by shred** [comment]
Reason for this hit is that GoldEd is invoking `intuition.library/SetPointer()` without intuition base loaded into register A6.

PatchWork assumes to find the library base in A6, and reads the library name from there for the hit output. However, PatchWork should not make assumptions on the contents of the registers. I am going to fix that issue.

**Comment by shred** [comment]
The reason for this MuForce hit was actually a bug in the "intuition/SetPointer/hot spot is outside of the sprite" check. The register values were incorrectly passed to the `ShowHit` function, which in turn showed invalid values in the PatchWork hit. This is fixed in commit fbe95683.

The MuForce hit was caused by `ShowHit` trying to deduce the library name based on the invalid contents of the `A6` register. Actually PatchWork should never assume that `A6` is actually pointing to the library base, even though it is required according to the RKRM. This is fixed in commit e036ee55.
